scripts/effects/effect-helpers.js

```diff
@@ -140,4 +140,64 @@ export default class EffectHelpers {
 
     return activeEffect.update({ statuses: statusesArray });
   }
+
+  /**
+   * Migrates the description of Custom Convenient Effects to use the ActiveEffect5e#description field instead of the legacy flags.
+   *
+   * @param {string} id - the ID of the custom effects item
+   * @returns {Promise} a promise that resolves when all the Item's ActiveEffects descriptions have migrated
+   */
+  async _migrateCustomEffectsItemEffectDescriptions(id, migration) {
+    const item = this._settings._findCustomEffectsItem(id);
+    const updates = [];
+    const scope = Constants.MODULE_ID;
+    const key_hasmigrated = Constants.FLAGS.HAS_MIGRATED;
+    for (const e of item.effects) {
+      if (e.getFlag(scope, key_hasmigrated) === migration) continue;
+      const effectDescription = e.description;
+      const flagDescription = e.getFlag(
+        Constants.MODULE_ID,
+        Constants.FLAGS.DESCRIPTION
+      );
+      const legacyDescription =
+        e.flags.convenientDescription ?? e.flags.customEffectDescription;
+      const key_description = Constants.FLAGS.DESCRIPTION;
+      const key_isConvenient = Constants.FLAGS.IS_CONVENIENT;
+      const description =
+        !!effectDescription &&
+        !['Applies Custom Effect', 'Applies custom effects', 'test'].some(
+          (d) => d === effectDescription
+        )
+          ? effectDescription
+          : !!flagDescription &&
+            !['Applies Custom Effect', 'Applies custom effects', 'test'].some(
+              (d) => d === flagDescription
+            )
+          ? flagDescription
+          : !!legacyDescription &&
+            !['Applies Custom Effect', 'Applies custom effects', 'test'].some(
+              (d) => d === legacyDescription
+            )
+          ? legacyDescription
+          : '';
+      updates.push({
+        _id: e.id,
+        description,
+        [`flags.${scope}.${key_isConvenient}`]: true,
+        [`flags.${scope}.${key_hasmigrated}`]: migration,
+        ['flags.-=isConvenient']: null, //remove remnants flags
```

Comment:
It's probably better to use the native function `unsetFlag` on an effect, rather than setting them to null.

Reply:
I wanted to make one batch update for each Item to get rid of all the relevant flags at once, instead of using the `unsetFlag` multiple times once for each `key`.

Reply:
Essentially using the code of `unsetFlag` function to unset all the flags in one go.
I can change this to `unsetFlag` and `setFlag` if needed.

Reply:
Oh, that's pretty funny that unsetFlag does exactly this. Okay then. Although you could have done something like using `await Promise.all(unsetFlagUpdates)` where `unsetFlagUpdates` is an array of promises from the `unsetFlag` function. But honestly that might introduce other issues, so this is fine.